messenger/multiple_buses.rst

```diff
@@ -0,0 +1,288 @@
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony
+                http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <!-- The bus that is going to be injected when injecting MessageBusInterface -->
+                <framework:messenger default-bus="messenger.bus.commands">
+                    <framework:bus name="messenger.bus.commands">
+                        <framework:middleware id="validation" />
+                        <framework:middleware id="doctrine_transaction" />
+                    <framework:bus>
+                    <framework:bus name="messenger.bus.queries">
+                        <framework:middleware id="validation" />
+                    <framework:bus>
+                    <framework:bus name="messenger.bus.events">
+                        <framework:middleware id="validation" />
+                        <framework:middleware id="allow_no_handler" />
+                    <framework:bus>
+                </framework:messenger>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/messenger.php
+        $container->loadFromExtension('framework', array(
+            'messenger' => array(
+                // The bus that is going to be injected when injecting MessageBusInterface
+                'default_bus' => 'messenger.bus.commands',
+                'buses' => array(
+                    'messenger.bus.commands' => array(
+                        'middleware' => array(
+                            'validation',
+                            'doctrine_transaction',
+                        ),
+                    ),
+                    'messenger.bus.queries' => array(
+                        'middleware' => array(
+                            'validation',
+                        ),
+                    ),
+                    'messenger.bus.events' => array(
+                        'middleware' => array(
+                            'validation',
+                            'allow_no_handler',
+                        ),
+                    ),
+                ),
+            ),
+        ));
+
+This will generate the ``messenger.bus.commands``, ``messenger.bus.queries``
+and ``messenger.bus.events`` services that you can inject in your services.
+
+Type-hints and Auto-wiring
+--------------------------
+
+Auto-wiring is a great feature that allows you to reduce the amount of configuration
+required for your service container to be created. When using multiple buses, by default,
+the auto-wiring will not work as it won't know which bus to inject in your own services.
```

Comment:
That's not totally true, right? The `messenger.bus.commands` service will be sent to you if you use the type-hint, by default. We should probably mention this.